### ISSUE TITLE
BART/T5 seq2seq example

### ISSUE THREAD

**Issue by deepanwayx** [huggingface/transformers#2847]
# 🚀 Feature request

Can we have a seq2seq example with training/fine-tuning and generation for BART/T5 models?

**Comment by sshleifer** [comment]
We are hard at work on this! I'd estimate 6 weeks out.

**Comment by SushantAI** [comment]
Looking forward to this for the T5 model :)

**Comment by adihaviv** [comment]
any updates?

**Comment by jeremysalwen** [comment]
The example doesn't seem to show training/fine-tuning, only evaluation of already fine-tuned models.

**Comment by morningmoni** [comment]
Hi @LysandreJik @sshleifer, as @jeremysalwen pointed out, the current example only involves the evaluation/inference of a fine-tuned model. Can we have a forward() function that supports seq2seq training with a "target_labels" parameter (like the "lm_labels" in existing models but measures loss in an auto-regressive manner)? Thanks for your great work!

**Comment by ArijRB** [comment]
@sshleifer Hello, any updates for training/fine-tuning on text generation for T5 model ?

**Comment by sshleifer** [comment]
`summarization/bart/finetune.py` supports T5.
